docs/developer-docs/integrations/sns/managing-sns/sns-asset-canister.md

```diff
@@ -0,0 +1,207 @@
+
+## Configuring an asset canister's permissions
+
+When configuring an asset canister, a set of permissions that contains a whitelist of principals must be created. This whitelist details who is allowed submit proposals that update assets in the asset canister. This whitelist must be configured prior to the SNS launch. Principals that are allowed to submit proposals are given the `Prepare` permission.
+
+Before the SNS can be launched, control of the asset canister must be handed over to the SNS. The SNS' governance canister should be added to the whitelist as a principal with `Commit` rights. Only principals with `Commit` rights may apply proposed changes. 
+
+Once the asset canister has been handed over to the SNS, only the governance canister should have `Commit` rights, and principals in the whitelist should have `Prepare` rights. The developer who configured and deployed the SNS should have their permissions removed prior to launching the SNS. 
+
+If the whitelist needs to be adapted or changed, the SNS can call `take_ownership` via a proposal, which must be added as a custom proposal type. This will clear all permissions, and give only the SNS governance canister `Commit` permissions. 
+
+### Granting permissions
+
+To grant a principal permission within an asset canister, the following command can be used:
+
+```
+dfx canister call --network ic <canister-id> grant_permission <principal>
+```
+
+### Revoking permissions
+
+To revoke a principal's permission within an asset canister, the following command can be used:
+
+```
+dfx canister call --network ic <canister-id>  revoke_permission <principal>
+```
+
+### Listing permissions
+
+To list permissions for an asset canister, the following command can be used:
+
+```
+dfx canister call --network ic <canister-id> list_permitted '(record {permission = variant {<Permission>}})'
+```
+
+For example, to list all principals with the `Commit` permission:
+
+```
+dfx canister call --network ic oa7fk-maaaa-aaaam-abgka-cai list_permitted '(record {permission = variant {Commit}})'
+```
+
+To list all principals with the `Prepare` permission:
+
+```
+dfx canister call --network ic oa7fk-maaaa-aaaam-abgka-cai list_permitted '(record {permission = variant {Commit}})'
+```
+
+Once the asset canister has been deployed and permissions have been configured, the SNS can be launched. You can learn more about launching an SNS [here](../launch-sns/launch-sns.md)
+
+## SNS GenericNervousSystemFunctions
+
+To submit a new `AddGenericNervousSystemFunction` SNS Proposal to support the `commit_proposed_batch` API, the target canister id should be the asset canister (that is upgraded in the upgrade steps below) and the target function is `commit_proposed_batch`. The validate function should be `validate_commit_proposed_batch`. 
+
+To submit an `ExecuteNervousSystemFunction` SNS Proposal with the output from `dfx deploy <frontend canister name> --network ic --by-proposal` see upgrade steps below.
```

Comment:
could add the info that this part is now defining how to execute the proposal that is now registered